File: src/main/java/xyz/mizarc/persistentitems/commands/PISubCommands/RemoveCommand.java
package xyz.mizarc.persistentitems.commands.PISubCommands;

import co.aikar.commands.BaseCommand;
import co.aikar.commands.annotation.CommandAlias;
import co.aikar.commands.annotation.Dependency;
import co.aikar.commands.annotation.Subcommand;
import org.bukkit.command.CommandSender;
import xyz.mizarc.persistentitems.ItemConfigIO;

@CommandAlias("pi")
public class RemoveCommand extends BaseCommand {

    @Dependency
    ItemConfigIO itemConfig;

    @Subcommand("remove")
    public void onRemove(CommandSender sender, String itemId) {
        if (!itemConfig.removeItem(itemId)) {
            sender.sendMessage("An item of that id doesn't exist");
            return;
        }
        sender.sendMessage("Item " + itemId + " has been removed");
    }
}

File: src/main/kotlin/dev/mizarc/itembinds/commands/persistentitems/RemoveCommand.kt
package dev.mizarc.itembinds.commands.persistentitems

import co.aikar.commands.annotation.*
import org.bukkit.command.CommandSender
import dev.mizarc.itembinds.commands.PersistentItemsCommand

@CommandAlias("persistentitems|pitems|pi")
class RemoveCommand : PersistentItemsCommand() {
    @Subcommand("remove")
    @CommandPermission("persistentitems.command.remove")
    @CommandCompletion("@pitems")
    @Syntax("<item>")
    fun onRemove(sender: CommandSender, name: String) {
        val item = itemRepo.getByName(name).firstOrNull()
        if (item == null) {
            sender.sendMessage("An item of that id doesn't exist.")
            return
        }

        itemRepo.remove(item)
        sender.sendMessage("Item $name has been removed.")
    }
}
